Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27320 Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend**

https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

xush6528 committed:
Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend

https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

[ghstack-poisoned]

satgera reviewed (approved):
Please ensure CI looks okay before landing.

xush6528 committed:
Update on "Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend"


https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

[ghstack-poisoned]

xush6528 committed:
Update on "Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend"


https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

[ghstack-poisoned]

xush6528 committed:
Update on "Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend"


https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

[ghstack-poisoned]

xush6528 commented:
- Add TestConfig type, so for other backends, it's clearer that it's fields have been modified,
- Format using BLACK.

xush6528 committed:
Update on "Make RpcTest re-usable by other RPC backends by using init_method to initialize a RPC backend"


https://github.com/pytorch/pytorch/pull/27208/

# Problem

Other RPC backends take init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17709219](https://our.internmc.facebook.com/intern/diff/D17709219/)

[ghstack-poisoned]
